pkg/plugin/v2/scaffolds/internal/templates/main.go

```diff
@@ -18,17 +18,14 @@ package templates
 
 import (
 	"fmt"
-	"hash/fnv"
 	"path/filepath"
-	"text/template"
```

Comment:
Why we are changing the v2+ plugin? 
This change requires to bump the controller-runtime and it has to break changes.  So, we can just to do it for v3+.

Reply:
We currently had copied these funcs into the v3 space and when I added support for the componentconfig implementation I was going to copy it to yet another but instead moved those to the `file/funcmap.go` so all templates can use this if they need https://github.com/kubernetes-sigs/kubebuilder/pull/1790/files#diff-13abe45cd80694321eb97754f6cb9aced5ccbbf51cd62b53baad70d4c4eff4e5R34-R41 I can revert this if we want the code should be the same just not needing a custom funcmap.

Reply:
I think it is fine 👍

Reply:
In order to keep it clean, I think this change should be done in a separate PR with the 🌱 tag, and this PR should be tagged v3-only, so that it is clear that the breaking changes are not happening to v2.

@camilamacedo86  @christopherhein

Reply:
:+1:

Reply:
I will remove this work after https://github.com/kubernetes-sigs/kubebuilder/pull/1812 is merged.